src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -1,44 +1,104 @@
-cd "$SOURCE_DIR"
-
 # Yarn config is per user, and since the build might run with a non-root account, we make sure
 # the yarn cache is set on every build.
 YARN_CACHE_DIR=/usr/local/share/yarn-cache
 if [ -d $YARN_CACHE_DIR ]
 then
 	echo
-    echo "Configuring Yarn cache folder ..."
+    echo "Configuring Yarn cache folder..."
     yarn config set cache-folder $YARN_CACHE_DIR
 fi
 
+allNodeModulesDirName=__oryx_all_node_modules
+if [ -d "$SOURCE_DIR"/$allNodeModulesDirName ]
+then
+	# Clean-up any earlier build artifacts (which contain only prod dependencies)
+	rm -rf node_modules
+	rm -f node_modules.zip
+
+	# Rename the folder which has all the node modules to reuse for later builds to improve perf
+	mv "$SOURCE_DIR"/$allNodeModulesDirName "$SOURCE_DIR"/node_modules
+fi
+
+# Directory where production dependencies are installed
```

Comment:
The logic here tries to install prod dependencies in a separate folder and then syncs up with the source folder where we run the full install. Finally we rename the node_modules folder having all dependencies to something else and copy the prod only dependencies from the separate folder back to source folder. This is done so that the base bash script code need not be touched and it does what it usually does.

However, I tried another approach which was simpler IMO but I hit a road block with it which I felt is risky. The approach is to restore prod dependencies in the source folder and instead use a separate folder for restoring all dependencies. The script looked less complex but the problem now was that the later commands like `npm run build` need to now know the custom node_modules folder having all the dependencies. This I fixed by using `NODE_PATH` which worked for `npm run build` but the same scenario did not work for `yarn run build`